seaworthy/logs.py

```diff
@@ -116,39 +143,84 @@ def __call__(self, line):
 
         return False
 
-    def __str__(self):
+    def args_str(self):
         matched = [str(m) for m in self._matchers[:self._position]]
         unmatched = [str(m) for m in self._matchers[self._position:]]
-        return 'SequentialLinesMatcher(matched=[{}], unmatched=[{}])'.format(
+        return 'matched=[{}], unmatched=[{}]'.format(
+            ', '.join(matched), ', '.join(unmatched))
+
+
+class AnyOrderLinesMatcher(LogMatcher):
```

Comment:
I was thinking a while ago of `OrderedLinesMatcher` and `UnorderedLinesMatcher`.

Reply:
I'm happy to change the names if you'd like. I do kind of prefer these to what we currently have.